test/powershell/Modules/Microsoft.PowerShell.Utility/Select-String.Tests.ps1

```diff
@@ -251,4 +253,46 @@ Describe "Select-String" -Tags "CI" {
             Select-String second $testInputFile -Raw -Context 2,2 | Should -BeExactly $expected
         }
     }
+
+    Context "Culture parameter" {
+        It "Should throw if -Culture parameter is used without -SimpleMatch parameter" {
+            { "1" | Select-String -Pattern "hello" -Culture "ru-RU" } | Should -Throw -ErrorId "CannotSpecifyCultureWithoutSimpleMatch,Microsoft.PowerShell.Commands.SelectStringCommand"
+        }
+
+        It "Should accept a culture: '<culture>'" -TestCases: @(
+            @{ culture = "Ordinal"},
+            @{ culture = "Invariant"},
+            @{ culture = "CurrentCulture"},
+            @{ culture = "ru-RU"}
+        ) {
+            param ($culture)
+            { "1" | Select-String -Pattern "hello" -Culture $culture -SimpleMatch } | Should -Not -Throw
+        }
+
+        It "Should works if -Culture parameter is a culture name" {
+            # Turkish has lower-case and upper-case version of the dotted "i",
+            # so the upper case of "i" (U+0069) isn't "I" (U+0049) but rather U+0130.
+            "file" | Select-String -Pattern "file" -Culture "tr-TR" -SimpleMatch | Should -BeExactly "file"
+            "file" | Select-String -Pattern "fIle" -Culture "tr-TR" -SimpleMatch | Should -BeNullOrEmpty
```

Comment:
Should you add a test with "`u{0130}" as a positive test case here?

Reply:
Seems like it would be better to have the -Culture variants as -TestCases

Reply:
Couldn't you do something like:

```powershell
It "...: <testcase>" -TestCases @(
  @{ testcase = '..."; pattern = '...'; culture = '...'; expected = '...' ; casesensitive = $true }
) {
 param ($pattern, $culture, $expected, $casesensitive)

'file' | Select-String -Pattern $pattern -Culture $culture -SimpleMatch -CaseSensitive:$casesensitive | Should -BeExactly $expected

Reply:
For the test case, thinking as black box testing instead of white box:

```powershell
'file' | Select-String -Pattern "f`u{0130}le" -Culture "tr-TR" -SimpleMatch | Should -BeExactly 'file'
```

Reply:
Done.